src/server/game/Entities/Pet/Pet.h

```diff
@@ -60,7 +60,7 @@ class Pet : public Guardian
     bool CreateBaseAtCreatureInfo(CreatureTemplate const* cinfo, Unit* owner);
     bool CreateBaseAtTamed(CreatureTemplate const* cinfo, Map* map, uint32 phaseMask);
     static std::pair<PetStable::PetInfo const*, PetSaveMode> GetLoadPetInfo(PetStable const& stable, uint32 petEntry, uint32 petnumber, bool current);
-    bool LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool current, uint32 healthPct = 0);
+    bool LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool current, uint32 healthPct = 0, bool fullMana = false);
```

Comment:
what about handling by pct like health

Reply:
I was thinking about that too. For the health parameter it makes sense because it can receive different values other than 100%, but in case of the mana it's either the last value or 100%, I don't think there is an scenario where a different value is needed, right?

Anyway, if you think it would be better to have it as a percentage, just in case, I'll change it no problem. let me know.